src/pke/include/scheme/ckksrns/ckksrns-schemeswitching.h

```diff
@@ -0,0 +1,420 @@
+
+    std::vector<Ciphertext<DCRTPoly>> EvalMaxSchemeSwitchingAlt(ConstCiphertext<DCRTPoly> ciphertext,
+                                                                PublicKey<DCRTPoly> publicKey, uint32_t numValues,
+                                                                uint32_t numSlots, bool oneHot, uint32_t pLWE,
+                                                                double scaleSign) override;
+
+    //------------------------------------------------------------------------------
+    // SERIALIZATION
+    //------------------------------------------------------------------------------
+
+    template <class Archive>
+    void save(Archive& ar) const {
+        ar(cereal::base_class<FHERNS>(this));
+    }
+
+    template <class Archive>
+    void load(Archive& ar) {
+        ar(cereal::base_class<FHERNS>(this));
+    }
+
+    std::string SerializedObjectName() const {
+        return "FHECKKSRNSSS";
+    }
+
+private:
+    //------------------------------------------------------------------------------
+    // Complex Plaintext Functions, copied from ckksrns-fhe. TODO: fix this
+    //------------------------------------------------------------------------------
+
+    Plaintext MakeAuxPlaintext(const CryptoContextImpl<DCRTPoly>& cc, const std::shared_ptr<ParmType> params,
+                               const std::vector<std::complex<double>>& value, size_t noiseScaleDeg, uint32_t level,
+                               usint slots) const;
+
+    Ciphertext<DCRTPoly> EvalMultExt(ConstCiphertext<DCRTPoly> ciphertext, ConstPlaintext plaintext) const;
+
+    void EvalAddExtInPlace(Ciphertext<DCRTPoly>& ciphertext1, ConstCiphertext<DCRTPoly> ciphertext2) const;
+
+    Ciphertext<DCRTPoly> EvalAddExt(ConstCiphertext<DCRTPoly> ciphertext1, ConstCiphertext<DCRTPoly> ciphertext2) const;
+
+    EvalKey<DCRTPoly> ConjugateKeyGen(const PrivateKey<DCRTPoly> privateKey) const;
+
+    Ciphertext<DCRTPoly> Conjugate(ConstCiphertext<DCRTPoly> ciphertext,
+                                   const std::map<usint, EvalKey<DCRTPoly>>& evalKeys) const;
+
+    /**
+   * Set modulus and recalculates the vector values to fit the modulus
+   *
+   * @param &vec input vector
+   * @param &bigValue big bound of the vector values.
+   * @param &modulus modulus to be set for vector.
+   */
+    void FitToNativeVector(uint32_t ringDim, const std::vector<int64_t>& vec, int64_t bigBound,
+                           NativeVector* nativeVec) const;
+
+#if NATIVEINT == 128 && !defined(__EMSCRIPTEN__)
+    /**
+   * Set modulus and recalculates the vector values to fit the modulus
+   *
+   * @param &vec input vector
+   * @param &bigValue big bound of the vector values.
+   * @param &modulus modulus to be set for vector.
+   */
+    void FitToNativeVector(uint32_t ringDim, const std::vector<__int128>& vec, __int128 bigBound,
+                           NativeVector* nativeVec) const;
+
+    constexpr __int128 Max128BitValue() const {
```

Comment:
remove Max128BitValue(), is128BitOverflow(double d), Max64BitValue(), and is64BitOverflow(double d) and use equivalent functions from [core/include/utils/utilities.h](https://github.com/openfheorg/openfhe-development/blob/b934174cdad23eaf8bfd7c8106df1b5e3ae8eeb6/src/core/include/utils/utilities.h#L111-L129)